allensdk/test/api/test_api.py

```diff
@@ -33,21 +33,51 @@
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 #
+
+import io
+from six.moves import builtins
+import zipfile
+import os
+
+import numpy as np
 import pytest
 from mock import MagicMock, patch, mock_open
-from allensdk.api.api import Api
-import allensdk.core.json_utilities as ju
 from requests.exceptions import HTTPError
-from six.moves import builtins
 import requests
 
+import allensdk.core.json_utilities as ju
+from allensdk.api.api import Api, stream_file_over_http, stream_zip_directory_over_http
+
+
 _msg = {'whatever': True}
 
 @pytest.fixture
 def api():
     return Api()
 
 
+@pytest.fixture
+def response():
+
+    resp = MagicMock()
+    resp.iter_content = lambda *a, **k: iter([b'1', b'2', b'3'])
+
+    return resp
+
+
+@pytest.fixture
+def zip_response():
+
+    flike = io.BytesIO()
+    data = '122333444455555'
+
+    zipper = zipfile.ZipFile(flike, mode='w')
+    zipper.writestr('test.txt', data)
+    zipper.close()
+
+    return flike.getvalue()
+
+
 @pytest.mark.xfail
```

Comment:
Does this test still need to be xfail?  It's probably been that way for a while, but we probably know enough about pytest to fix that now.